src/Symfony/Component/DependencyInjection/ContainerAware.php

```diff
@@ -17,6 +17,8 @@
  * @author Fabien Potencier <fabien@symfony.com>
  *
  * @api
+ * 
+ * @deprecated Deprecated since version 2.7, to be removed in 3.0. Use ContainerAwareTrait with ContainerAwareInterface instead.
```

Comment:
I think deprecating things for which better alternatives have been found is a good thing. In the long run having many ways to do things will confuse people, complicate documentation etc. If this one is deprecated, people will see themselves what the recommended approach is. Just think about all the different syntaxes to reference resources in symfony and the confusion and long-standing issue because of that.